app/models/payment_intent.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+class PaymentIntent < ApplicationRecord
+  belongs_to :holder, polymorphic: true
+  belongs_to :payment_record, polymorphic: true
+  belongs_to :initiated_by, class_name: 'User' # For now only users can initiate payments - in future, this may become a polymorphic association
+  belongs_to :confirmation_source, polymorphic: true, optional: true
+  belongs_to :cancellation_source, polymorphic: true, optional: true
+
+  validate :valid_status_combination
+
+  scope :unpaid, -> { where.not(wca_status: 'succeeded') }
+  scope :started, -> { where.not(wca_status: 'created') }
+
+  # TODO: Refactor this or move it into this class
+  delegate :stripe_id, :stripe_status, :parameters, :money_amount, :find_account_id, to: :payment_record
+
+  # Stripe secrets are case-sensitive. Make sure that this information is not lost during encryption.
+  encrypts :client_secret, downcase: false
+
+  serialize :error_details, coder: JSON
+
+  enum wca_status: {
+    created: 'created', # A record has been created on the payment provider's system (no payment action yet initiated by the user)
+    pending: 'pending', # The user is now attempting to pay, but has not reached a completion state yet
+    partial: 'partial', # Some but not all funds have been paid (eg, if the user has selected an instalment-based payment option)
+    failed: 'failed', # The payment did not succeed for any reason (insufficient funds, user error) but the user can still try to pay
+    succeeded: 'succeeded', # Completion state - The full amount due is confirmed as being paid by the payment provider
+    canceled: 'canceled', # Completion state - the user has indicated that they will no longer attempt to complete payment
+  }
+
+  def confirmed?
+    !self.confirmed_at.nil?
+  end
+
+  def started?
+    self.wca_status != 'created'
```

Comment:
Fun fact about Rails enums: Because you defined `created` as an enum status, you can access `self.started?` for free. Rails adds the question-mark method as a convenience for you (same applies to `some_intent.failed?`, `some_intent.succeeded?` and similar)

Reply:
Updated

Reply:
Is it though? In my most recent diff on GitHub, this still shows as `self.wca_status != 'created'`, but my point is you can just do `!self.created?`

Reply:
It is now actually updated :halo: